git/git_cached_repository.bzl

```diff
@@ -0,0 +1,222 @@
+load("//git:git_cached_worker.bzl", "checkout", "clone","fetch")
+
+"""Rules for caching external git repositories in favor of keeping a lighter git network overhead."""
+
+load(
+    "@bazel_tools//tools/build_defs/repo:utils.bzl",
+    "patch",
+    "update_attrs",
+    "workspace_and_buildfile",
+)
+
+_env_var_enable_verbosity = "GIT_CACHED_VERBOSE"
+
+_common_attrs = {
+    "remote_url":
+        attr.string(
+            mandatory=True,
+            doc="The URI of the remote Git repository",
+        ),
+    "commit":
+        attr.string(
+            mandatory=True,
+            doc="Specific commit to be checked out",
+        ),
+    "branch":
+        attr.string(
+            default="master",
+            doc="Specify a branch to fetch HEAD references",
+        ),
+    "shallow_since":
+        attr.string(
+            default="",
+            doc="An optional date in addition to a specified commit. " +
+            "Setting such a date close to the " +
+            "specified commit allows for a more shallow clone of the " +
+            "repository, saving bandwidth and wall-clock time.",
+        ),
+    "cache_directory":
+        attr.string(
+            default="{}/.git-cache/{}",
```

Comment:
seems like a template that without values, is it okay?

Reply:
It is the default value being used to store the git cache in `$HOME/.git-cache/<name-of-the repo>`.
It is string formatted only if no explicit value supplied.